.pre-commit-config.yaml

```diff
@@ -0,0 +1,16 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: check-yaml
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
```

Comment:
The trailing-whitespace thing is going to be a problem because some of the test suite data deliberately relies on this.  Maybe adding an ignore list explicitly, if that's possible.